### PR TITLE
[WIP] UI/UX Community upgrade [Issue #1762]

### PR DESCRIPTION
**Pull request by wackerow**
## Description
Starting with very rough draft to get discussion going regarding an upgraded **/community/** page.

Goals (*copied from issue):
- Re-design `/community/` section into a `community.js` page-conditional, instead of using `/content/community/index.md`
- Include some new graphics and potentially give this a theme inline with new Eth2 section changes
- UX that makes buttons to places like GitHub, Reddit, Discord, Meetups far more obvious, prominent and enticing/inviting
- Potentially include improved way for users to add events, and automatically remove expired events
- Potentially include social media widgets
- Most importantly, wanted to open the discussion for more ideas here

image
_*updated screenshot as of a2cc9c6_
_(note: image very much roughly put together as well just to demonstrate a different placeholder)_

My first thought was to re-populate the current `.md` data into this `.js` file, and play around with some layout options. Will post concepts as they come, but would ❤️ any thoughts/suggestions! 

### Direct [Netlify /en/community](https://deploy-preview-1763--ethereumorg.netlify.app/en/community/) link for this PR

### CODE REVIEW

**Comment by wackerow** [comment]
### Update as of 6ae0e63
@ryancreatescopy @samajammin I have kept everything in one file for now, but plan is to start breaking this apart soon to declutter this community "home page." _Known issue with mobile sized screen since `<CardList />` implementation; will address._

Again, these banner images were initially made as placeholders. Let me know if there is anything preferable as a substitute, or if you like where these are going just let me know what adjustments you may want since these were not necessarily intended to be final draft images (I still have the vector files).

I made the custom social media platform buttons to give a standard look across the site. My thought was this could potentially be broken off into a reusable component in other places on the site as well to continue with the theming. These are intended to brighten up content and guide people toward places they quickly recognize visually, that they may otherwise miss buried in all-text lists (Again, I have the vector files of these buttons and they can all be adjusted if desired, and happy to create more).

Let me know your thoughts!

**Comment by ryancreatescopy** [comment]
Hey @wackerow thanks for all your work. Personally, I find the social media images a little too distracting. I'd prefer us to use icons (I think they're all accessible using our Icon component and https://react-icons.github.io/react-icons/). 

I've not had time to properly review your PR yet but just wanted to share some more background info about this page's goals and some user feedback I found in our backlog: https://www.notion.so/efdn/Community-v-2-0-e3984a5a28124e4ab95bfc59cdc3c072

Will get to taking a proper look when I can.

**Comment by wackerow** [comment]
@ryancreatescopy only one I don't see (of the ones so far) is ETH Global (which makes sense). 
Thanks for mentioning this, didn't think react-icons had all these, and can definitely re-design this without these PNGs. 
My plan was originally to drop these as a draft and potentially tune them for light/dark, but this will definitely be easier (less querying/loading too). Was hoping to add some color via these graphics to help attract people.

(This is just from my own experience, but I had personally been to this page on multiple occasions in the past, and feel like I missed many of the gems listed because it required "reading" the page to see it all, and I often would just glance.... vs being able to spot familiar color-themes/icons quickly to help guide users ie. twitter blue, reddit orange, etc)

File: src/pages-conditional/community.js [hunk 1]
```diff
@@ -0,0 +1,906 @@
...
+  },
+  {
+    title: "Ethereum Research",
+    to: "https://ethresear.ch",
+    description:
+      "The most influential messageboard for cryptoeconomic research",
+  },
+]
+
+const events = [
```
Comment (samajammin) [inline src/pages-conditional/community.js:389]: One idea to address [the concern](https://github.com/ethereum/ethereum-org-website/issues/1762#issuecomment-725652852) that moving from Markdown --> JS will make this harder to contribute - perhaps we move this & other source data into the src/data/ directory as a standalone CSV/YAML/JS file, so it's more apparent how to make edits/additions to this information. Might be a good approach to separate the data from the app logic, so e.g. we could have events, jobs, communities, meetups all sourced within /data/.

Thoughts?

**Comment by samajammin** [comment]
Agree re: use of icons vs. custom images for both UX & performance.

File: src/pages-conditional/community.js (hunk 1 above)
Comment (wackerow) [reply src/pages-conditional/community.js:389]: haha, just saw this after replying to your other note... I agree, 100% think these should go into their own files to make editing very straight forward

**Comment by samajammin** [comment]
@wackerow overall this is coming together nicely! This will be a strong improvement over the existing page. A few thoughts:
- Meetup search: given that this is a large list, I do like the idea of a search feature to filter results, however, I'm not convinced search should be the _only_ way to find the meetup list. By hiding results until someone searches, we're hiding this content from search engines. It's also not obvious to the user what they are searching for (name? location?) or what the results will be. I'd advocate we A) display the full list in a table, as it currently exists or B) display the full table by default with the search filter above that filters the table results.
- Online communities: The list UI feels a bit loud to me with all the different background colors. I also think the existing categorization of forums vs chat rooms vs others is a valuable separation to quickly find what format someone might be most interested in. I'd like to keep that categorization.
- Before you put too much time into more changes, I'd like our illustrious content designer @ryancreatescopy to think through this layout & provide some direction. Hopefully he'll have capacity to explore this in the next couple weeks but no promises 😄

**Comment by samajammin** [comment]
Closing this out for now - let's re-open this when we revisit this as a priority.

For context, see our current roadmap here: https://github.com/ethereum/ethereum-org-website/issues/2284

**Pull request closed without merging**